pkg/tfbridge/provider.go

```diff
@@ -1013,15 +1009,20 @@ func (p *Provider) Create(ctx context.Context, req *pulumirpc.CreateRequest) (*p
 	label := fmt.Sprintf("%s.Create(%s/%s)", p.label(), urn, res.TFName)
 	glog.V(9).Infof("%s executing", label)
 
+	props, err := plugin.UnmarshalProperties(req.GetProperties(),
+		plugin.MarshalOptions{Label: label, KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshaling %s's new property state", urn)
+	}
 	// To get Terraform to create a new resource, the ID must be blank and existing state must be empty (since the
 	// resource does not exist yet), and the diff object should have no old state and all of the new state.
-	config, assets, err := UnmarshalTerraformConfig(ctx,
-		p, req.GetProperties(), res.TF.Schema(), res.Schema.Fields,
-		fmt.Sprintf("%s.news", label))
+	inputs, assets, err := makeTerraformInputsNoDefaultsWithMaxItemsOneDefaults(
```

Comment:
It feels like we have reached the point where we should replace `makeTerraformInputsHelper` with 
```go
makeTerraformInputsWithOptions(..., makeTerraformInputOptions{
	Defaults: False(),
	MaxItemsOneDefaults: True(),
})
```

`makeTerraformInputsNoDefaultsWithMaxItemsOneDefaults` is pretty long and hard to read. I first read it as `makeTerraformInputsNoDefaultsWithMaxItemsOneNoDefaults`.

We would then fold `makeTerraformInputsWithoutTFDefaults` into the same scheme.

Reply:
If we want to name these functions, let's name them something that describes where they should be used, not just the flags that we have set. For example, we can express this excellent comment (borrowed from a test case) like this:

```go
// missing MaxItems=1 properties should present to TF as empty collections when creating
// a resource (makeTerraformInputsNoDefaultsWithMaxItemsOneDefaults)
```

```go
// makeTerraformInputsForResourceCreate translates Pulumi property values 
// to the form that TF expects for a resource creation.
func makeTerraformInputsForResourceCreate(...) ... {
	return makeTerraformInputsWithOptions(..., makeTerraformInputOptions{
	Defaults: False(),
	MaxItemsOneDefaults: True(),
}
```

Reply:
I've refactored the makTerraformInputs* as suggested, LMK if it now makes more sense.

Reply:
Looks good to me.